Documentation/guides/drivers.rst

```diff
@@ -0,0 +1,211 @@
+.. include:: /substitutions.rst
+.. _drivers:
+
+Drivers
+=======
+
+Some NuttX boards don't have full support for all the on-chip peripherals. If you need support for this hardware,
+you will either need to port a driver from another chip, or write one yourself. This section discusses how to do that.
+
+Porting a Driver
+----------------
+
+Often support for on-chip peripherals exists in a closely related chip, or even a different family or from a different
+manufacturer. Many chips are made up of different Intellectual Property (IP) blocks that are licensed from vendors like
+Cadence, Synopsys, and others. The IP blocks may be similar enough to use another chip's driver with little
+modification.
+
+* Find a similar driver in NuttX source code:
+
+  * Look at register names listed in the datasheet for the peripheral.
+  * Search the NuttX codebase for the register names (try several different ones).
+  * Note that you'll have to compare the datasheet to the header and code files to see if there are differences; there
+    will usually be some differences between architectures, and they can be significant.
+
+* Find a similar driver in U-Boot source code:
+
+  * Only for inspiration, you can't copy code because of license incompatibility.
+  * But you can debug to see how the driver works.
+  * `U-Boot <https://www.denx.de/wiki/U-Boot>`_ drivers are often easier to understand than BSD Unix drivers because
+    U-Boot is simpler.
+
+* Find a similar driver in Zephyr or BSD Unix (OpenBSD, FreeBSD, NetBSD):
+
+  * If you find one, you can borrow code directly (Apache 2.0 and BSD licenses are compatible).
```

Comment:
I think we are trying to keep all 3rd party code out.

Reply:
@btashton Even Apache License 2.0 code?

Reply:
Yes otherwise it cannot carry ASF header it must retain the original one.  So it is Apache 2.0 compatible, but it still must refer to the third party license unless there is a SGA.

Reply:
@btashton I'll take this section out then. But then what good is having compatible open source licenses then? It seems like that negates a huge benefit of projects collaborating and exchanging code.

Reply:
It is license compatible, for instance we could decided to include a 3rd party Apache 2.0 compression library and the resulting binary would be under Apache 2.0 but we just cannot say that code is ASF code.

Reply:
@btashton Ok, took it out, and stated that code from other projects can only be used for inspiration, not copied, unless there is an SGA for it on file from the authors.